addons/web_editor/static/src/js/wysiwyg/widgets/select_options_dialog.js

```diff
@@ -0,0 +1,59 @@
+odoo.define('wysiwyg.widgets.SelectOptionsDialog', function (require) {
+'use strict';
+
+const core = require('web.core');
+const Dialog = require('wysiwyg.widgets.Dialog');
+
+const _t = core._t;
+
+const SelectOptionsDialog = Dialog.extend({
+    template: 'wysiwyg.widgets.selectoptions',
+    xmlDependencies: Dialog.prototype.xmlDependencies.concat(
+        ['/web_editor/static/src/xml/wysiwyg.xml']
+    ),
+    events: _.extend({}, Dialog.prototype.events, {
+        'change textarea.o_select_options': '_onChangeOptions',
+    }),
+
+    /**
+     * @constructor
+     */
+    init: function (parent, options, select) {
+        this._super(parent, _.extend({}, {
+            title: _t("Edit your options"),
+            save_text: _t("Add"),
+        }, options || {}));
+        this.select = select;
+        if (this.select) {
+            this.selectOptions = [...this.select.querySelectorAll('option')].map(option => {
+                return $(option).text();
+            });
+        } else {
+            this.selectOptions = [];
+        }
+    },
+    /**
+     * @override
+     */
+    save: function () {
+        const $options = this.selectOptions.map(option => $(`<option>${option}</option>`));
+        const $select = $('<select contenteditable="false"/>');
+        $select.append(...$options);
+        this.final_data = $select[0];
+        return this._super(...arguments);
+    },
+    selectOptionsText: function () {
+        return this.selectOptions.join('\n');
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    _onChangeOptions: function (ev) {
+        this.selectOptions = $(ev.target).val().split('\n').map(option => option.trim());
```

Comment:
Can this handle options with multiple break line in the text arear. 
ie : 
```
option1
option2


option3

option4
```
This will probably generate some emption option.
maybe we could split using "smarter" regex.

Reply:
I could just do `.split(/\n+/)` but I assumed if people leave a blank space it's because they actually do want a blank option (for formatting).